server/runtime/src/main/java/org/infinispan/server/security/realm/EncryptedPropertiesSecurityRealm.java

```diff
@@ -0,0 +1,500 @@
+
+         return this;
+      }
+
+      /**
+       * Set the {@link InputStream} to use to load the users.
+       *
+       * @param usersStream the {@link InputStream} to use to load the users.
+       * @return this {@link Builder}
+       */
+      public Builder setUsersStream(InputStream usersStream) {
+         this.usersStream = usersStream;
+
+         return this;
+      }
+
+      /**
+       * Set the {@link InputStream} to use to load the group information.
+       *
+       * @param groupsStream the {@link InputStream} to use to load the group information.
+       * @return this {@link Builder}
+       */
+      public Builder setGroupsStream(InputStream groupsStream) {
+         this.groupsStream = groupsStream;
+
+         return this;
+      }
+
+      /**
+       * Where this realm returns an {@link AuthorizationIdentity} set the key on the Attributes that will be used to
+       * hold the group membership information.
+       *
+       * @param groupsAttribute the key on the Attributes that will be used to hold the group membership information.
+       * @return this {@link Builder}
+       */
+      public Builder setGroupsAttribute(final String groupsAttribute) {
+         this.groupsAttribute = groupsAttribute;
+
+         return this;
+      }
+
+
+      /**
+       * Set the default realm name to use if no realm name is discovered in the properties file.
+       *
+       * @param defaultRealm the default realm name if one is not discovered in the properties file.
+       * @return this {@link Builder}
+       */
+      public Builder setDefaultRealm(String defaultRealm) {
+         this.defaultRealm = defaultRealm;
+
+         return this;
+      }
+
+      /**
+       * Set format of users property file - if the passwords are stored in plain text. Otherwise is HEX( MD5( username
+       * ":" realm ":" password ) ) expected.
+       *
+       * @param plainText if the passwords are stored in plain text.
+       * @return this {@link Builder}
+       */
+      public Builder setPlainText(boolean plainText) {
+         this.plainText = plainText;
+
+         return this;
+      }
+
+      /**
+       * Builds the {@link EncryptedPropertiesSecurityRealm}.
+       *
+       * @return built {@link EncryptedPropertiesSecurityRealm}
+       * @throws IOException                   when loading of property files fails
+       * @throws java.io.FileNotFoundException when property file does not exist
+       * @throws RealmUnavailableException     when property file of users does not contain realm name specification
+       */
+      public EncryptedPropertiesSecurityRealm build() throws IOException {
+         EncryptedPropertiesSecurityRealm realm = new EncryptedPropertiesSecurityRealm(this);
+         realm.load(usersStream, groupsStream);
+
+         return realm;
+      }
+
+   }
+
+   private static class LoadedState {
+
+      private final Map<String, AccountEntry> accounts;
+      private final String realmName;
```

Comment:
`realmName` is never used.

Reply:
I'd rather keep it around in case we may want to use it. It's code that comes from Elytron's original implementation.